### PR TITLE
[lang] Remove redundant TensorType expansion for function returns

### PR DESCRIPTION
**Pull request by jim19930609**
Issue: https://github.com/taichi-dev/taichi/issues/5819

### CODE REVIEW

File: taichi/transforms/type_check.cpp [hunk 1]
```diff
@@ -196,9 +196,17 @@ class TypeCheck : public IRVisitor {
   }
 
   void visit(UnaryOpStmt *stmt) override {
-    stmt->ret_type = stmt->operand->ret_type;
+    auto operand_type = stmt->operand->ret_type;
```
Comment (strongoier) [inline taichi/transforms/type_check.cpp:199]: Are these changes related to this PR?

Comment (jim19930609) [reply taichi/transforms/type_check.cpp:199]: Yup, we do have some test cases with `real_matrix_scalarize=False` such as `test_matrix_return_real_matrix`. These tests used to work because of the scalarization at Python level, so after removing the `expand_exprs()` we'll have to fix the issues with CodeGen and runtime.

**Review by strongoier — APPROVED** [review]
